apps/sms/views/shared/js/utils.js

```diff
@@ -711,6 +711,60 @@
     },
 
     /**
+     * Returns a rate limited function, useful to reduce the number of function
+     * calls when listening to user input (keypress) or browser events.
+     * @params {function} func - The function that should be rate-limited
+     * @params {number} delay - Min amount of time (ms) between two calls
+     * @param {Boolean} preventfirstCall - Prevent first synchronous call when
+     * true.
+     * @param {Boolean} preventLastCall - Prevent last trailing call when true.
+     * @returns {function} - The rate limited function that will call the
+     *  original function
+     */
+    throttle: function(func, delay, preventfirstCall, preventLastCall) {
+      var args;
+      var timeout = null;
+      var previous = 0;
+
+      if(typeof func !== 'function'){
+        throw new Error('func must be a Function');
+      }
+
+      if(typeof delay !== 'number' || delay < 0){
+        throw new Error('delay must be a positive number');
+      }
+
+      return function() {
+        var now = Date.now();
+        if (!previous && preventfirstCall){
+          previous = now;
+        }
+        var remaining = delay - (now - previous);
+        args = arguments;
+        if (remaining <= 0 || remaining > delay) {
+          if (timeout) {
+            clearTimeout(timeout);
+            timeout = null;
+          }
+          previous = now;
+          func.apply(this, args);
```

Comment:
I could use `func(...arguments)` here but not on line 758 where the apply call is wrapped in a setTimeout + `arguments` is a pseudo array so I would have to use Array.from(arguments) before spreading it.
I tried to use the rest parameter on function in line 737 and then use it along with the spread operator but it seems to break all the unit test (I suspect because it is launch from node).
What do you think ?